packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseSecuritySchemeObject-test.js

```diff
@@ -0,0 +1,357 @@
+      });
+
+      const parseResult = parse(context, securityScheme);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult).to.contain.warning("'Security Scheme Object' is missing required property 'in'");
+    });
+
+    it('provides a warning when in is not a string', () => {
+      const securityScheme = new namespace.elements.Object({
+        type: 'apiKey',
+        name: 'example',
+        in: 1,
+      });
+
+      const parseResult = parse(context, securityScheme);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult).to.contain.warning("'Security Scheme Object' 'in' is not a string");
+    });
+
+    it('provides a warning when in is not a permitted value', () => {
+      const securityScheme = new namespace.elements.Object({
+        type: 'apiKey',
+        name: 'example',
+        in: 'space',
+      });
+
+      const parseResult = parse(context, securityScheme);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult).to.contain.warning("'Security Scheme Object' 'in' must be either 'query', 'header' or 'cookie'");
+    });
+  });
+
+  describe('#scheme', () => {
+    it('provides a warning when scheme does not exist', () => {
+      const securityScheme = new namespace.elements.Object({
+        type: 'http',
+      });
+
+      const parseResult = parse(context, securityScheme);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult).to.contain.warning("'Security Scheme Object' is missing required property 'scheme'");
+    });
+
+    it('provides a warning when scheme is not a string', () => {
+      const securityScheme = new namespace.elements.Object({
+        type: 'http',
+        scheme: 1,
+      });
+
+      const parseResult = parse(context, securityScheme);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult).to.contain.warning("'Security Scheme Object' 'scheme' is not a string");
+    });
+  });
+
+  describe('#description', () => {
+    it('attaches description to member', () => {
+      const securityScheme = new namespace.elements.Object({
+        type: 'apiKey',
+        name: 'example',
+        in: 'query',
+        description: 'an example security scheme',
+      });
+
+      const parseResult = parse(context, securityScheme);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult.get(0)).to.be.instanceof(namespace.elements.AuthScheme);
+      expect(parseResult.get(0).description.toValue()).to.equal(
+        'an example security scheme'
+      );
+    });
+
+    it('provides a warning when description is not a string', () => {
+      const securityScheme = new namespace.elements.Object({
+        type: 'apiKey',
+        name: 'example',
+        in: 'query',
+        description: true,
+      });
+
+      const parseResult = parse(context, securityScheme);
+
+      expect(parseResult).to.contain.warning("'Security Scheme Object' 'description' is not a string");
```

Comment:
I wonder if the parseResult here contains the security scheme or not because this warning is not essential.